### ISSUE TITLE
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: number. Check the render method of Home. 

### ISSUE THREAD

**Issue by fatcatt** [software-mansion/react-native-svg#2358]
# Question
**environment：**
react native: 0.72.7
xcode: Version 15.4
 react-native-svg: "^15.4.0",
react-native-svg-transformer: "^1.5.0",

my **metro.config.js** config
<img width="955" alt="截屏2024-07-20 17 24 34" src="https://github.com/user-attachments/assets/b8eabc6d-4df0-4892-844c-52cfa0b61a6d">

and my error tips:
![9801721467515_ pic](https://github.com/user-attachments/assets/ccb63394-ceb5-4f29-b717-1ab5a6e8ca94)

I observed that a similar issue has been raised two weeks ago, but the issue has been closed and I did not find the exact solution there. Has anyone encountered a similar situation? How should we solve it, thanks

**Comment by bohdanprog** [comment]
Hello @fatcatt,
Could you please provide a simple example to reproduce the issue you refer to?
Additionally, could you share more details about the issue you are talking?

Thank you.

**Comment by lukachi** [comment]
Expo environment

devDeps:
"react-native-svg": "15.2.0",
"react-native-svg-transformer": "^1.5.0",

metro.config.js
```js
const { getDefaultConfig } = require("expo/metro-config");
const { withNativeWind } = require('nativewind/metro');

const withSvgTransformer = (config) => {
  const { resolver, transformer } = config;

  config.transformer = {
    ...transformer,
    babelTransformerPath: require.resolve("react-native-svg-transformer/react-native")
  };
  config.resolver = {
    ...resolver,
    assetExts: resolver.assetExts.filter((ext) => ext !== "svg"),
    sourceExts: [...resolver.sourceExts, "svg"]
  };

  return config;
}

module.exports = (() => {
  let config = getDefaultConfig(__dirname, { isCSSEnabled: true })

  config = withSvgTransformer(config)
  // config = withCircomFilesAndPolyfills(config)

  config = withNativeWind(config, { input: './src/theme/global.css' })

  return config
})()
```

usage:
```ts
import WarningIcon from '@assets/icons/warning-icon.svg'

<WarningIcon />
```

results in:
```
 (NOBRIDGE) ERROR  Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
```

but sometimes, after several app reloads - it start to work, but only once

**Comment by bohdanprog** [comment]
Hello @lukachi,
Can you please provide a full example of that problem?
Thank you.

**Comment by lukachi** [comment]
Hi @bohdanprog , which details should i provide?

Also want to note, that i'm using newArch

**Comment by bohdanprog** [comment]
@lukachi, it would be great if you could create a repository with your example app. 
This will help us in resolving the issue.

Thank you.

**Comment by lukachi** [comment]
sure, 

[here](https://github.com/lukachi/rn-template/blob/migrate/new-arch/src/ui/UiIcon.tsx) is the repo i'm currently working on

but now i've got [this](https://stackoverflow.com/questions/78494804/expo-stuck-on-welcome-to-expo-screen) error after updating to new expo sdk :) so, the things just have got worse)

currently keeping an eye on [this](https://github.com/expo/expo/issues/28680) issue discussion

**Comment by trojanyao** [comment]
I encountered the same issue when using `<Filter>`.

<img width="1529" alt="Xnip2024-07-31_15-59-59" src="https://github.com/user-attachments/assets/7a9b6ed3-f891-48d3-8c54-8ae0ede8a0d5">
<img width="1534" alt="Xnip2024-07-31_16-00-17" src="https://github.com/user-attachments/assets/8104f10a-65fc-4d99-9594-dcf4293c1828">

- react-native: 0.73.6
- react-native-svg: 15.4.0

**Comment by bohdanprog** [comment]
@lukachi, 
here is a screen from my app that I set up.
![Screenshot_1722412894](https://github.com/user-attachments/assets/ca2ae224-e16c-4b77-a5d5-48726e915952)
here is an [example](https://github.com/bohdanprog/react-native-svg-import-icon)

**Comment by jakex7** [comment]
> I encountered the same issue when using `<Filter>`.
> 
> * react-native: 0.73.6
> * react-native-svg: 15.4.0

Hi @trojanyao, are you using expo go by any chance? If yes, then you'd have to use prebuilt or wait for Expo team to update the `react-native-svg` package in the app.
Otherwise, if you're not using expo, please create the new issue and place your minimal reproduction repository there.

**Comment by trojanyao** [comment]
@jakex7 Ya, I'm using Expo. Thanks for pointing that out. I got it.

**Comment by lukachi** [comment]
@bohdanprog Hi again, sorry for late respond

i've migrated my app from expo-router to 'react-navigation/native', so app starts fine now.
Could you please run my app from `migrate/new-arch` branch?

There you can see, that imported `svg` files become an empty object, and also if `svg` files will be imported and used straightfully - it will cause the error in this issue topic, but after reload app several times, there will be a chance that it starts to work, but once through several reloads

https://github.com/lukachi/rn-template/blob/migrate/new-arch/src/ui/UiIcon.tsx

**Comment by bohdanprog** [comment]
Hi @lukachi :)
Can you give me more context on whether its problem appears in android or ios, new or old arch?

**Comment by lukachi** [comment]
Hi @bohdanprog it's finally works

For me, the solution was requiring components instead of import, idk why :)

and also i've used 
    `babelTransformerPath: require.resolve("react-native-svg-transformer")`
instead of 
    `babelTransformerPath: require.resolve("react-native-svg-transformer/react-native")`
or
    `babelTransformerPath: require.resolve("react-native-svg-transformer/expo")`

here is commit: https://github.com/lukachi/rn-template/commit/05e17182ca2ca4f8363c0decd2f39fe596a698de

**Comment by bohdanprog** [comment]
Hi @lukachi :)

I'm glad to hear that! 👍 I believe it depends on how the project is configured, similar to how I set it up in my [example](https://github.com/bohdanprog/react-native-svg-import-icon/blob/main/metro.config.js#L9). 
Please feel free to reach out to us. 
We would be delighted to help you in the future. 😉

**Comment by bohdanprog** [comment]
@fatcatt Can we close that issue?
Or do you still have some problem with that?

**Comment by lukachi** [comment]
thank u @bohdanprog 

I think we can close this issue
but it would be great if in docs would be several examples for different cases like that.

**Comment by bohdanprog** [comment]
@lukachi
The issue wasn't with react-native-svg but with the configuration of react-native-svg-transformer.
Perhaps I overlooked something.
